consensus/core/src/dag_state.rs

```diff
@@ -103,45 +103,45 @@ impl DagState {
 
         let mut unscored_committed_subdags = Vec::new();
 
-        let last_committed_rounds = {
-            let commit_info = store
-                .read_last_commit_info()
-                .unwrap_or_else(|e| panic!("Failed to read from storage: {:?}", e));
+        let commit_info = store
+            .read_last_commit_info()
+            .unwrap_or_else(|e| panic!("Failed to read from storage: {:?}", e));
+        let (mut last_committed_rounds, commit_recovery_start_index) =
             if let Some((commit_ref, commit_info)) = commit_info {
-                let mut committed_rounds = commit_info.committed_rounds;
-                let last_commit = last_commit
-                    .as_ref()
-                    .expect("There exists commit info, so the last commit should exist as well.");
-
-                if last_commit.index() > commit_ref.index {
-                    let committed_blocks = store
-                        .scan_commits(((commit_ref.index + 1)..last_commit.index() + 1).into())
-                        .unwrap_or_else(|e| panic!("Failed to read from storage: {:?}", e))
-                        .iter()
-                        .flat_map(|commit| {
-                            let committed_subdag =
-                                load_committed_subdag_from_store(store.as_ref(), commit.clone());
-                            if context
-                                .protocol_config
-                                .mysticeti_leader_scoring_and_schedule()
-                            {
-                                unscored_committed_subdags.push(committed_subdag.clone());
-                            }
-                            committed_subdag.blocks
-                        })
-                        .collect::<Vec<_>>();
-
-                    for block in committed_blocks {
-                        committed_rounds[block.author()] =
-                            max(committed_rounds[block.author()], block.round());
+                (commit_info.committed_rounds, commit_ref.index + 1)
```

Comment:
Log commit_info and commit_ref in this case? And maybe log if they are not found too.